### ISSUE TITLE
Voiceover focuses incorrectly when opening a menu inside a menubar

### ISSUE THREAD

**Issue by btisdall** [ariakit/ariakit#5262]
### Current behavior

With a menu embedded in a menu bar, opening the menu per Voiceover's guidance causes focus to jump out of the menu bar.

Note that the reproduction code uses the [MenubarProvider example](https://ariakit.org/reference/menubar-provider) from the docs.

See also [video](https://github.com/user-attachments/assets/59d4cc71-6e2f-496c-a043-5a1b101d1693)

Tested using macOS. Reproduces on at least Chrome, Safari and Firefox.

### Steps to reproduce the bug

1. Open https://stackblitz.com/edit/github-kstw55tx-drd1v3vk?file=src%2Fapp.tsx
2. Open preview in new window
3. Enable Voiceover
4. Navigate into the web content area then into the menubar and onto the first button
5. Follow the prompt to open the "File" menu
6. Observe that the menu opens but the Voiceover focus moves out of the web content area and that navigating the menu items using ctrl-option-leftarrow does not work

### Expected behavior

1. Open https://stackblitz.com/edit/github-kstw55tx-drd1v3vk?file=src%2Fapp.tsx
2. Open preview in new window
3. Enable Voiceover
4. Navigate into the web content area then into the menubar and onto the first button
5. Follow the prompt to open the "File" menu
6. On opening the menu, Voiceover focus should be on the menu and navigating the menu items using ctrl-option-leftarrow and ctrl-option-rightarrow should work.

**Comment by diegohaz** [comment]
This looks like a VoiceOver/Chrome bug. I couldn’t reproduce it in Safari. If you can, please upload a video and let me know which versions of macOS and which browser versions you’re using.

**Comment by diegohaz** [comment]
> I included a video for macOS 26.1 (25B78)/Chrome (143.0.7499.193): https://github.com/user-attachments/assets/59d4cc71-6e2f-496c-a043-5a1b101d1693

Thanks for sharing, but that's the same video you attached to the original issue, right? I can reproduce it in Chrome, and it looks like a browser bug. I can't reproduce it in Safari 26.0, so it may have been introduced in Safari 26.1.

**Comment by diegohaz** [comment]
@btisdall Thanks! I’ll look into what changed in those versions that might be causing this issue.

**Comment by diegohaz** [comment]
I can't reproduce this in Safari 26.4, so I guess they might have fixed it?

I tested https://ariakit.com/components/menubar
